mech_client/marketplace_interact.py

```diff
@@ -296,10 +297,12 @@ def send_marketplace_request(  # pylint: disable=too-many-arguments,too-many-loc
     :type marketplace_contract: Web3Contract
     :param gas_limit: Gas limit.
     :type gas_limit: int
-    :param prompt: The request prompt.
-    :type prompt: str
-    :param tool: The requested tool.
-    :type tool: str
+    :param prompts: The request prompts.
+    :type prompts: tuple
+    :param tools: The requested tools.
+    :type tools: tuple
+    :param num_requests: The total number of requests.
+    :type num_requests: int
```

Comment:
Not sure if it's worth passing instead of recalculating. It's a cheap operation.